newzealidar/s3_connection.py

```diff
@@ -0,0 +1,217 @@
+                    except AttributeError:
+                        # Extract and convert dataset's CRS spatial reference to EPSG code
+                        epsg_code = CRS.from_string(data.spatial_ref.crs_wkt).to_epsg()
+                    # Write the EPSG code as the Coordinate Reference System (CRS) for the dataset
+                    data.rio.write_crs(epsg_code, inplace=True)
+        # If the file extension is neither ".pickle" nor ".nc"
+        else:
+            # Read the content of the retrieved object using geopandas
+            data = gpd.read_file(BytesIO(body))
+        # Log a message confirming successful retrieval from the S3 bucket
+        log.info(f"Successfully retrieved '{s3_object_key}' from the S3 bucket.")
+        return data
+
+    def remove_object(self, s3_object_key: Union[str, pathlib.Path]) -> None:
+        """
+        Removes an object from the S3 bucket.
+        Parameters
+        ----------
+        s3_object_key : Union[str, pathlib.Path]
+            The key of the object to be removed from the S3 bucket. If a pathlib.Path object is provided,
+            it will be converted to a string representation.
+        """
+        # Check if the provided s3_object_key is a pathlib.Path object
+        if isinstance(s3_object_key, pathlib.Path):
+            # Convert the pathlib.Path object to a string representation
+            s3_object_key = s3_object_key.as_posix()
+        # Delete the object with the provided object key from the S3 bucket
+        self.s3_client.delete_object(Bucket=self.bucket_name, Key=s3_object_key)
+        # Log a message confirming successful deletion from the S3 bucket
+        log.info(f"Successfully deleted '{s3_object_key}' from the S3 bucket.")
+
+    def store_file(self, s3_object_key: Union[str, pathlib.Path], file_path: Union[str, pathlib.Path]) -> None:
+        """
+        Uploads a file to the S3 bucket.
+        Parameters
+        ----------
+        s3_object_key : Union[str, pathlib.Path]
+            The key under which to store the file in the S3 bucket. If a pathlib.Path object is provided,
+            it will be converted to a string representation.
+        file_path : Union[str, pathlib.Path]
+            The local file path of the file to be uploaded.
+        """
+        # Check if the provided s3_object_key is a pathlib.Path object
+        if isinstance(s3_object_key, pathlib.Path):
+            # Convert the pathlib.Path object to a string representation
+            s3_object_key = s3_object_key.as_posix()
+        # Upload the file at 'file_path' to the S3 bucket with the provided object key
+        self.s3_client.upload_file(Bucket=self.bucket_name, Key=s3_object_key, Filename=file_path)
+        # Log a message confirming successful storage in the S3 bucket
+        log.info(f"Successfully stored `{s3_object_key}` in the S3 bucket.")
+
+    def retrieve_file(self, s3_object_key, file_path):
```

Comment:
```suggestion
    def retrieve_file(self, s3_object_key: Union[str, pathlib.Path], file_path: Union[str, pathlib.Path]) -> None:
```

Reply:
Good spotting. Have it in the docstring but forgot to add it here.